geoips/schema/sectors_dynamic.yaml

```diff
@@ -0,0 +1,60 @@
+type: object
+required:
+  - interface
+  - metadata
+  - spec
+properties:
+  interface:
+    type: string
+  metadata:
+    type: object
+    required: [name, family, short-description, long-description]
+    properties:
+      name:
+        type: string
+      family:
+        type: string
+      short-description:
+        type: string
+      long-description:
+        type: string
+  spec:
+    type: object
+    additionalProperties: false # Ensure top level fields are consistent

```

Comment:
I updated the schema accordingly for the new YAML layout:

spec:
  area_def_generator:
    name:
    arguments:
      .......

Oddly, when I accidentally had duplicate "area_def_generator" fields, it still passed validation.  I don't know why that is, and would expect it to fail if the spec contained 2 area_def_generator fields (and the first had no name or arguments defined - which I would think would be required).

If someone could please take a closer look at this schema to make sure it is assembled correctly, I would appreciate it.

Reply:
I think I'll create an issue for this to be handled another time. The issue is that, when pyyaml loads the file, it doesn't error when it encounters duplicate fields. It just takes the last one it encounters. So, this isn't a validation failure, this is a reading failure...

For now, we can just avoid duplicate fields.

Reply:
See #162